php/php.editor/src/org/netbeans/modules/php/editor/codegen/AutoImport.java

```diff
@@ -0,0 +1,910 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.php.editor.codegen;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.text.BadLocationException;
+import javax.swing.text.Document;
+import org.netbeans.api.annotations.common.CheckForNull;
+import org.netbeans.api.annotations.common.NullAllowed;
+import org.netbeans.api.editor.document.LineDocumentUtils;
+import org.netbeans.api.lexer.Token;
+import org.netbeans.api.lexer.TokenSequence;
+import org.netbeans.api.lexer.TokenUtilities;
+import org.netbeans.editor.BaseDocument;
+import org.netbeans.modules.csl.api.EditList;
+import org.netbeans.modules.csl.api.OffsetRange;
+import org.netbeans.modules.editor.indent.api.IndentUtils;
+import org.netbeans.modules.php.api.util.StringUtils;
+import org.netbeans.modules.php.editor.CodeUtils;
+import org.netbeans.modules.php.editor.api.AliasedName;
+import org.netbeans.modules.php.editor.api.QualifiedName;
+import org.netbeans.modules.php.editor.indent.CodeStyle;
+import org.netbeans.modules.php.editor.lexer.LexUtilities;
+import org.netbeans.modules.php.editor.lexer.PHPTokenId;
+import org.netbeans.modules.php.editor.model.GroupUseScope;
+import org.netbeans.modules.php.editor.model.ModelElement;
+import org.netbeans.modules.php.editor.model.ModelUtils;
+import org.netbeans.modules.php.editor.model.NamespaceScope;
+import org.netbeans.modules.php.editor.model.UseScope;
+import static org.netbeans.modules.php.editor.model.UseScope.Type.CONST;
+import static org.netbeans.modules.php.editor.model.UseScope.Type.FUNCTION;
+import static org.netbeans.modules.php.editor.model.UseScope.Type.TYPE;
+import org.netbeans.modules.php.editor.parser.PHPParseResult;
+import org.netbeans.modules.php.editor.parser.astnodes.DeclareStatement;
+import org.netbeans.modules.php.editor.parser.astnodes.EmptyStatement;
+import org.netbeans.modules.php.editor.parser.astnodes.Program;
+import org.netbeans.modules.php.editor.parser.astnodes.Statement;
+import org.netbeans.modules.php.editor.parser.astnodes.visitors.DefaultVisitor;
+
+public final class AutoImport {
+
+    public static final String PARAM_NAME = "php-auto-import"; // NOI18N
+    public static final String PARAM_KEY_FQ_NAME = "fqName"; // NOI18N
+    public static final String PARAM_KEY_ALIAS_NAME = "aliasName"; // NOI18N
+    public static final String PARAM_KEY_USE_TYPE = "useType"; // NOI18N
+    public static final String USE_TYPE = "type"; // NOI18N
+    public static final String USE_FUNCTION = "function"; // NOI18N
+    public static final String USE_CONST = "const"; // NOI18N
+    private static final Logger LOGGER = Logger.getLogger(AutoImport.class.getName());
```

Comment:
Nitpick: Don't we put `LOGGER` always as the first field in a class?

Reply:
Will do that :) Thanks!

Reply:
Fixed.